WasatchNET/EEPROMJSON.cs

```diff
@@ -0,0 +1,425 @@
+        public int ActualPixelsHoriz;
+        public int ActivePixelsHoriz;
+        public int ActivePixelsVert;
+        public int MinIntegrationTimeMS;
+        public int MaxIntegrationTimeMS;
+        public int ROIHorizStart;
+        public int ROIHorizEnd;
+        public int[] ROIVertRegionStarts;
+        public int[] ROIVertRegionEnds;
+        public double[] LaserPowerCoeffs;
+        public double MaxLaserPowerMW;
+        public double MinLaserPowerMW;
+        public double ExcitationWavelengthNM;
+        public double AvgResolution;
+        public int[] BadPixels;
+        public string UserText;
+        public string ProductConfig;
+        public int RelIntCorrOrder;
+        public double[] RelIntCorrCoeffs;
+        public bool Bin2x2;
+        public bool FlipXAxis;
+
+        public override bool Equals(object obj)
+        {
+            var item = obj as EEPROMJSON;
+
+            //public string Serial;
+            if (item.Serial != this.Serial)
+                return false;
+            //public string Model;
+            if (item.Model != this.Model)
+                return false;
+            //public int SlitWidth;
+            if (item.SlitWidth != this.SlitWidth)
+                return false;
+            //public double BaudRate;
+            if (Math.Abs(item.BaudRate - this.BaudRate) > 0.00001)
+                return false;
+            //public bool IncBattery;
+            if (item.IncBattery != this.IncBattery)
+                return false;
+            //public bool IncCooling;
+            if (item.IncCooling != this.IncCooling)
+                return false;
+            //public bool IncLaser;
+            if (item.IncLaser != this.IncLaser)
+                return false;
+            //public int StartupIntTimeMS;
+            if (item.StartupIntTimeMS != this.StartupIntTimeMS)
+                return false;
+            //public int StartupTempC;
+            if (item.StartupTempC != this.StartupTempC)
+                return false;
+            //public int StartupTriggerMode;
+            if (item.StartupTriggerMode != this.StartupTriggerMode)
+                return false;
+            //public double DetectorGain;
+            if (Math.Abs(item.DetectorGain - this.DetectorGain) > 0.00001)
+                return false;
+            //public double DetectorGainOdd;
+            if (Math.Abs(item.DetectorGainOdd - this.DetectorGainOdd) > 0.00001)
+                return false;
+            //public double DetectorOffset;
+            if (item.DetectorOffset != this.DetectorOffset)
+                return false;
+            //public double DetectorOffsetOdd;
+            if (item.DetectorOffsetOdd != this.DetectorOffsetOdd)
+                return false;
+            //public double[] WavecalCoeffs;
+            if (item.WavecalCoeffs.Length != this.WavecalCoeffs.Length)
+                return false;
+            for (int i = 0; i < this.WavecalCoeffs.Length; ++i)
```

Comment:
and in fact a bool floatEq(float[] a, float[] b, thresh = 0.00001)  (where the first thing it does is return false if a.Length != b.Length)

Reply:
This and all other fixes should be in now